tests/app/UnitTests/GitCommands.Tests/GitModuleTests_Remotes.cs

```diff
@@ -0,0 +1,282 @@
+                Assert.AreEqual(1, remotes[0].PushUrls.Count);
+                Assert.AreEqual("git://github.com/RussKie/gitextensions.git", remotes[0].PushUrls[0]);
+
+                Assert.AreEqual("origin", remotes[1].Name);
+                Assert.AreEqual("git@github.com:drewnoakes/gitextensions.git", remotes[1].FetchUrl);
+                Assert.AreEqual(1, remotes[1].PushUrls.Count);
+                Assert.AreEqual("git@github.com:drewnoakes/gitextensions.git", remotes[1].PushUrls[0]);
+
+                Assert.AreEqual("upstream", remotes[2].Name);
+                Assert.AreEqual("git@github.com:gitextensions/gitextensions.git", remotes[2].FetchUrl);
+                Assert.AreEqual(1, remotes[2].PushUrls.Count);
+                Assert.AreEqual("git@github.com:gitextensions/gitextensions.git", remotes[2].PushUrls[0]);
+
+                Assert.AreEqual("asymmetrical", remotes[3].Name);
+                Assert.AreEqual("https://github.com/gitextensions/fetch.git", remotes[3].FetchUrl);
+                Assert.AreEqual(1, remotes[3].PushUrls.Count);
+                Assert.AreEqual("https://github.com/gitextensions/push.git", remotes[3].PushUrls[0]);
+
+                Assert.AreEqual("with-space", remotes[4].Name);
+                Assert.AreEqual("c:/Bare Repo", remotes[4].FetchUrl);
+                Assert.AreEqual(1, remotes[4].PushUrls.Count);
+                Assert.AreEqual("c:/Bare Repo", remotes[4].PushUrls[0]);
+
+                Assert.AreEqual("multi", remotes[5].Name);
+                Assert.AreEqual("git@github.com:drewnoakes/gitextensions.git", remotes[5].FetchUrl);
+                Assert.AreEqual(2, remotes[5].PushUrls.Count);
+                Assert.AreEqual("git@github.com:drewnoakes/gitextensions.git", remotes[5].PushUrls[0]);
+                Assert.AreEqual("git@gitlab.com:drewnoakes/gitextensions.git", remotes[5].PushUrls[1]);
+
+                Assert.AreEqual("with_option", remotes[6].Name);
+                Assert.AreEqual("https://github.com/flannelhead/jsmn-stream.git", remotes[6].FetchUrl);
+                Assert.AreEqual(1, remotes[6].PushUrls.Count);
+                Assert.AreEqual("https://github.com/flannelhead/jsmn-stream.git", remotes[6].PushUrls[0]);
+            }
+        });
+    }
+
+    [Test]
+    public void GetRemoteNames()
+    {
+        string[] lines = new[] { "RussKie", "origin", "upstream", "asymmetrical", "with-space" };
+
+        using (_executable.StageOutput("remote", string.Join("\n", lines)))
+        {
+            IReadOnlyList<string> remotes = _gitModule.GetRemoteNames();
+
+            Assert.AreEqual(lines, remotes);
+        }
+    }
+
+    [Test]
+    public void GetRemoteColors_should_return_saved_colors()
+    {
+        using GitModuleTestHelper moduleTestHelper = new();
+        GitModule gitModule = GetGitModuleWithExecutable(_executable, module: moduleTestHelper.Module);
```

Comment:
AI output must be checked thrice! When resolving the MCs, it took long until I noticed that this code makes use of `_executable` but not of `_gitModule`.

Reply:
Yep, some of these tests are special...